src/graph/graph.c

```diff
@@ -290,287 +124,291 @@ Graph *Graph_Get(RedisModuleCtx *ctx, RedisModuleString *graph_name) {
     return g;
 }
 
+// Return number of nodes graph can contain.
+size_t Graph_NodeCap(const Graph *g) {
+    return g->nodes->itemCap;
+}
+
 size_t Graph_NodeCount(const Graph *g) {
     assert(g);
     return g->nodes->itemCount;
 }
 
+// All graph matrices are required to be squared NXN
+// where N is Graph_RequiredMatrixDim.
+size_t Graph_RequiredMatrixDim(const Graph *g) {
+    // Matrix dimensions should be at least:
+    // Number of nodes + number of deleted nodes.
+    return g->nodes->itemCount + array_len(g->nodes->deletedIdx);
```

Comment:
`array_len` expands into a fairly complex expression - we should check sometime to make sure that the compiler's capable of optimizing it. If not, I wouldn't mind storing this value.

At present, I don't think I like the idea of adding buffer space. We have 1 matrix for every label and 2 for every relation type - the memory overhead could grow really dramatically.

Reply:
The other alternative I had in mind had a much larger memory overhead,
As this current implementation kills bulk insert performance, as with each labeled node we're inserting we need to perform resize before labeling the next node.